### PR TITLE
fix: [PLG-2299]: handling toggle tobe blocked when premium support is on 

### PR DESCRIPTION
**Pull request by dhruvUpadhyay07**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/27-platform/auth-settings/components/Subscription/PaymentMethod/PaymentMethodStep.tsx [hunk 1]
```diff
@@ -151,6 +151,7 @@ export default function PaymentMethodStep({
                 paymentMethodInfo={subscriptionProps.paymentMethodInfo}
                 setView={setView}
                 module={module}
+                fromPaymentMethodPage={true}
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/components/Subscription/PaymentMethod/PaymentMethodStep.tsx:154]: NIT: `true` is auto-inferred. Can simply pass `fromPaymentMethodPage` as a prop

File: src/modules/27-platform/auth-settings/components/Subscription/PricePreview/PricePreview.tsx [hunk 2]
```diff
@@ -180,7 +181,7 @@ const PricePreview: React.FC<PricePreviewProps> = ({
       }
     })?.unitAmount
   )
-
+  const isPremuiumSupported = subscriptionProps?.premiumSupport || false
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/components/Subscription/PricePreview/PricePreview.tsx:184]:
```suggestion
  const isPremiumSupported = subscriptionProps?.premiumSupport || false
```

File: src/modules/27-platform/auth-settings/components/Subscription/PricePreview/PricePreview.tsx [hunk 3]
```diff
@@ -236,14 +237,13 @@ const PricePreview: React.FC<PricePreviewProps> = ({
...
   }
-
   return (
     <Layout.Vertical className={cx(css.pricePreview, colorBorder)}>
       <Text font={{ variation: FontVariation.H4 }} padding={{ bottom: 'large' }}>
         {getString('platform.authSettings.pricePreview.title')}
       </Text>
       <PaymentFrequencyToggle
-        disabled={!canChangePaymentFrequency}
+        disabled={!canChangePaymentFrequency || isPremuiumSupported}
```
Comment (mayankVermaHarness) [inline src/modules/27-platform/auth-settings/components/Subscription/PricePreview/PricePreview.tsx:246]:
```suggestion
        disabled={!canChangePaymentFrequency || isPremiumSupported}
```

**Review by mayankVermaHarness — APPROVED** [review]

**Comment by mayankVermaHarness** [comment]
@dhruvUpadhyay07 Please add Description and Screenshots in PRs

**Review by satyam-harness — APPROVED** [review]

**Pull request merged**